Spec/RealtimeClientChannel.swift

```diff
@@ -1118,6 +1111,76 @@ class RealtimeClientChannel: QuickSpec {
                     }
                 }
 
+                // RTL5i
+                it("if the channel is in a pending state DETACHING, do the detach operation after the completion of the pending request") {
+                    let options = AblyTests.commonAppSetup()
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    let channel = client.channels.get("foo")
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.attach() { error in
+                            expect(error).to(beNil())
+                            done()
+                        }
+                    }
+
+                    var detachedCount = 1
+                    channel.on(.Detached) { stateChange in
+                        expect(stateChange?.reason).to(beNil())
+                        detachedCount += 1
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.once(.Detaching) { stateChange in
+                            guard let stateChange = stateChange else {
+                                fail("ChannelStateChange is nil"); done(); return
+                            }
+                            expect(stateChange.reason).to(beNil())
+                            expect(stateChange.current).to(equal(ARTRealtimeChannelState.Detaching))
+                            expect(stateChange.previous).to(equal(ARTRealtimeChannelState.Attached))
+                            channel.detach()
+                            done()
+                        }
+                        channel.detach()
+                    }
+
+                    expect(detachedCount) == 1
+
+                    channel.off()
+                }
+
+                // RTL5i
+                it("if the channel is in a pending state ATTACHING, do the detach operation after the completion of the pending request") {
+                    let options = AblyTests.commonAppSetup()
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    let channel = client.channels.get("foo")
+
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+                        channel.once(.Attaching) { stateChange in
```

Comment:
Surely you need to check that the state moves to the `ATTACHED` state i.e. "do the detach operation after the completion of the pending request".  The pending operation is to attach the channel